service/worker/replicator/processor.go

```diff
@@ -437,6 +435,22 @@ RetryLoop:
 		}
 		break RetryLoop
 	}
+
+	if retryErr, ok := err.(*shared.RetryTaskError); ok {
+		beginRunID := retryErr.GetRunId()
+		beginEventID := retryErr.GetNextEventId()
+		endRunID := attr.GetRunId()
+		endEventID := attr.GetFirstEventId()
+
+		// history service will return RetryTaskError, we can use it as an hint
+		errResend := p.historyRereplicator.SendMultiWorkflowHistory(
```

Comment:
This feature needs to be rolled out to all clusters before enabling it.  Here are the 2 options I see:
1) If target cluster is running old version I expect something like handler not found error.  We should disable reties on this error and proceed with old style processing.
2) Separate config knob to enable/disable this feature.  Config knob is enabled once feature is rolled out to all clusters.  Config knob will be useful in other situations too, like disabling this feature due to unforeseen issue.